skpro/regression/linear/_glm.py

```diff
@@ -201,6 +332,88 @@ def __init__(
         self.max_start_irls = max_start_irls
         self.add_constant = add_constant
 
+        if family == "0":
```

Comment:
this looks excessive - have you looked at the deprecation handling? It has examples at the end.

I think we should avoid setting new defaults for unaffected parameters.